store/modules/stimulation/getters.js

```diff
@@ -41,18 +39,22 @@ export default {
   },
 };
 
-const get_protocol_editor_color = (list) => {
-  const color = generate_random_color(false);
-  const duplicate_color = list.filter((protocol) => protocol.color === color).length > 0;
-
-  return duplicate_color ? get_protocol_editor_color(list) : color;
-};
-
 // TODO Luci, handle if there are more than 26 protocols
 const get_protocol_editor_letter = (list) => {
-  const current_protocol_assignment = list[list.length - 1].letter;
-  const current_alphabet_idx = alphabet.indexOf(current_protocol_assignment);
-  return alphabet[current_alphabet_idx + 1];
+  // grab just the first letter in case of AA BB CCC DDDD, unlikely but just in case.
+  const current_protocol_assignment = list[list.length - 1].letter[0];
+
+  const current_alphabet_idx = ALPHABET.indexOf(current_protocol_assignment);
+  let letter_assignment = ALPHABET[current_alphabet_idx + 1];
+
+  if (current_alphabet_idx === 25) {
+    letter_assignment = ALPHABET[0];
+  }
+  // add double/triple/etc letters based on number of protocols so it doesn't assign same single letter to protocols
+  let assign = "";
+  const num_of_loops = [...Array(list.length).keys()].filter((x) => x % 26 === 0);
```

Comment:
Could you just divide the length of the list by 26 and chop off the remainder to get this number?

Reply:
I initially had `Math.floor(x / 26)` but it results in `X`, `Y`, `ZZ` in the first round
`Math.floor(x / 27)` results in `YY`, `ZZ`,  `AA` in the second round

Reply:
Something like this should work
```
foo = (n) => {
    let idx = n % 26;
    let label = String.fromCharCode(65 + idx);
    let num_additions = Math.floor(n / 26);
    for (let i = 0; i < num_additions; i++) {
        label += String.fromCharCode(65 + idx);
    }
    return label;
}
```
Here's what it outputs
```
> Array.from({length: 100}, (x, n) => n).map(n => foo(n))
[
  'A',    'B',    'C',    'D',    'E',    'F',    'G',    'H',
  'I',    'J',    'K',    'L',    'M',    'N',    'O',    'P',
  'Q',    'R',    'S',    'T',    'U',    'V',    'W',    'X',
  'Y',    'Z',    'AA',   'BB',   'CC',   'DD',   'EE',   'FF',
  'GG',   'HH',   'II',   'JJ',   'KK',   'LL',   'MM',   'NN',
  'OO',   'PP',   'QQ',   'RR',   'SS',   'TT',   'UU',   'VV',
  'WW',   'XX',   'YY',   'ZZ',   'AAA',  'BBB',  'CCC',  'DDD',
  'EEE',  'FFF',  'GGG',  'HHH',  'III',  'JJJ',  'KKK',  'LLL',
  'MMM',  'NNN',  'OOO',  'PPP',  'QQQ',  'RRR',  'SSS',  'TTT',
  'UUU',  'VVV',  'WWW',  'XXX',  'YYY',  'ZZZ',  'AAAA', 'BBBB',
  'CCCC', 'DDDD', 'EEEE', 'FFFF', 'GGGG', 'HHHH', 'IIII', 'JJJJ',
  'KKKK', 'LLLL', 'MMMM', 'NNNN', 'OOOO', 'PPPP', 'QQQQ', 'RRRR',
  'SSSS', 'TTTT', 'UUUU', 'VVVV'
]
```

Reply:
I can put that, I feel like that's more lines of code than what's currently being used.
`[...Array(60).keys()].map((n)=> { 
   let idx = n % 26; let label = String.fromCharCode(65 + idx); 
   let num_additions = Math.floor(n / 26); 
   for (let i = 0; i < num_additions; i++) { label += String.fromCharCode(65 + idx); } 
   return label; 
 }` vs.
 
 `  const num_of_loops = [...Array(list.length).keys()].filter((x) => x % 26 === 0);
[...Array(num_of_loops.length)].map(() => (assign += letter_assignment));
 `

Reply:
That was just an example all wrapped up into a single function to make it easier to demonstrate. The main part I wanted to show was this
```
let num_additions = Math.floor(n / 26);
for (let i = 0; i < num_additions; i++) {
  label += String.fromCharCode(65 + idx);
}
```

You could incorporate it into the actual code like this
```
const get_protocol_editor_letter = list => {
  const protocol_idx = list.length - 1;

  const letter_assignment = ALPHABET[protocol_idx % 26];
  const num_letters = Math.floor(protocol_idx / 26) + 1;

  // I changed the snippet above to this here since I just found out this exists
  return letter_assignment.repeat(num_letters);
};
```
Which ends up being less lines of code overall and is more readable (less visual noise, intent more clear).

I just tested it through `AAAA` and it seems to be working